### DISCUSSION TITLE
println alignment issue

### DISCUSSION THREAD

**Discussion by tbillion** [Bodmer/TFT_eSPI#1945] [General]
```

  tft.setCursor(15, 45);
  tft.setTextColor(TFT_CYAN , TFT_BLACK);
  tft.setTextSize(1);
  tft.println(debug_info);

```

i am confused why when i use this code the first line prints in the correct spot as if indented and the line that follows that begins on column zero is there a way to use the print line based on a custom column location? 
![20220801_190838](https://user-images.githubusercontent.com/16172731/182265290-10fa0c1c-7d93-4532-9860-9aef3f208355.jpg)

i in the simple context of whats happening have no control over the blurb of text in the array, i guess if there is no way to do this i could take the long way around and make a helper function.

**Comment by Frank-Bemelman** [comment]
You could set a viewport at 15,45,w,h
SetViewPort(...

**Comment by Bodmer** [comment]
There is no built in capability to define the x coordinate for a "carraige return" so this will have to be managed in the sketch.

Frank's suggestion of using setViewPort(...) is an option that seems to fit in with your use case.

**Comment by tbillion** [comment]
Thanks guys works like a charm



On Tue, Aug 2, 2022, 1:51 PM Bodmer ***@***.***> wrote:

> There is no built in capability to define the x coordinate for a "carraige
> return" so this will have to be managed in the sketch.
>
> Frank's suggestion of using setViewPort(...) is an option that seems to
> fit in with your use case.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Bodmer/TFT_eSPI/discussions/1945#discussioncomment-3310542>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD3MNOZGJNU4IJCF5JDGSN3VXFU37ANCNFSM55JIXAMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
![20220802_173140](https://user-images.githubusercontent.com/16172731/182484998-6dc93600-c1ce-41f3-b848-3bc00cb39b63.jpg)
